modules/polygon/src/polygon-utilities.d.ts

```diff
@@ -0,0 +1,80 @@
+
+type SegmentVisitor = (p1: number[], p2: number[], i1: number, i2: number) => void;
+
+type SegmentVisitorFlat = (p1x: number, p1y: number, p2x: number, p2y: number, i1: number, i2: number) => void;
```

Comment:
Would be nice to have only one visitor type (the flat one), though I suppose it would be a breaking change.